src/optimalcontrol/tests/WorhpInterfaceTest.cpp

```diff
@@ -0,0 +1,227 @@
+        jacobian.resize(2,2);
+        jacobian(0,0) = 1.0;
+        jacobian(1,1) = 1.0;
+        return true;
+    } //using dense matrices, but the sparsity pattern is still obtained
+
+    virtual bool evaluateConstraintsHessian(const iDynTree::VectorDynSize& constraintsMultipliers, iDynTree::MatrixDynSize& hessian) override {
+        hessian.resize(2,2);
+        hessian.zero();
+        return true;
+    } //using dense matrices, but the sparsity pattern is still obtained
+
+    bool setPlusInfinity(double plusInfinity) {
+        if (plusInfinity < 0)
+            return false;
+
+        m_plusInfinity = plusInfinity;
+        return true;
+    }
+
+    bool setMinusInfinity(double minusInfinity) {
+        if (minusInfinity > 0)
+            return false;
+
+        m_minusInfinity = minusInfinity;
+        return true;
+    }
+
+    double expectedMinimum() {
+        double expected;
+        if (m_c1 < (-m_b1/(2*m_a1)))
+            expected = - m_b1 * m_b1 / (4*m_a1);
+        else
+            expected = m_a1*m_c1*m_c1 + m_b1*m_c1;
+
+        if (m_c2 < (-m_b2/(2*m_a2)))
+            expected += - m_b2 * m_b2 / (4*m_a2);
+        else
+            expected += m_a2*m_c2*m_c2 + m_b2*m_c2;
+
+        return expected;
+    }
+
+    const iDynTree::VectorDynSize& expectedVariables() {
+        m_expectedVariables.resize(2);
+        if (m_c1 < (-m_b1/(2*m_a1)))
+            m_expectedVariables(0) = - m_b1 / (2*m_a1);
+        else
+            m_expectedVariables(0) = m_c1;
+
+        if (m_c2 < (-m_b2/(2*m_a2)))
+            m_expectedVariables(1) = - m_b2 / (2*m_a2);
+        else
+            m_expectedVariables(1) = m_c2;
+
+        return m_expectedVariables;
+    }
+};
+
+int main(){
+    iDynTree::optimization::WorhpInterface worhpSolver;
+    std::shared_ptr<TestProblem> problem(new TestProblem);
+    iDynTree::VectorDynSize guess(2), dummy1, dummy2, dummy3;
+    guess.zero();
+
+    ASSERT_IS_TRUE(worhpSolver.setWorhpParam("AcceptTolOpti", 1e-6));
+    ASSERT_IS_TRUE(worhpSolver.setWorhpParam("AcceptTolFeas", 1e-6));
+    ASSERT_IS_TRUE(worhpSolver.setWorhpParam("Algorithm", 2));
+    ASSERT_IS_TRUE(worhpSolver.setWorhpParam("NLPprint", -1));
+
+
+    ASSERT_IS_TRUE(problem->setMinusInfinity(worhpSolver.minusInfinity()));
+    ASSERT_IS_TRUE(problem->setPlusInfinity(worhpSolver.plusInfinity()));
+    ASSERT_IS_TRUE(worhpSolver.setProblem(problem));
+    for (int i = 0; i < 5; ++i){
+        ASSERT_IS_TRUE(worhpSolver.solve());
+        double optimalCost;
+        ASSERT_IS_TRUE(worhpSolver.getOptimalCost(optimalCost));
+        ASSERT_EQUAL_DOUBLE_TOL(optimalCost, problem->expectedMinimum(), 1e-5);
+        iDynTree::VectorDynSize solution;
+        ASSERT_IS_TRUE(worhpSolver.getPrimalVariables(solution));
+        ASSERT_EQUAL_VECTOR_TOL(solution, problem->expectedVariables(), 1e-5);
+        ASSERT_IS_TRUE(worhpSolver.getDualVariables(dummy1, dummy2, dummy3));
+    }
```

Comment:
There seems to be a lot of duplication between the tests for the optimizer interfaces. Have you considered a way to consolidate this (just a suggestion, this is not blocking the PR).

Reply:
Yes you are right, there is a lot of duplication. For the time being I stick to simple copypastas to have freedom from case to case. Between different solvers there may be specific methods (usually for the settings) and this represents the main difference between tests. Also, I may try different things depending on the type of solver. Somehow I liked the fact that each test is "atomic", but as the number of interfaces grows I agree that the maintenance of the tests would suffer.